apps/wallet-mobile/src/features/ReviewTx/common/hooks/useFormattedMetadata.tsx

```diff
@@ -42,16 +42,14 @@ export const formatMetadata = async (
   }
 }
 
-const parseMsg = (msg: Array<string>) => {
-  if (msg.length > 1) {
-    const message = msg.join('')
-    try {
-      return JSON.parse(message)
-    } catch {
-      return message
-    }
+const parseMsg = (msg: Array<string> | string): string => {
```

Comment:
@michaeljscript -> I know what is bugging me, the return type can be whatever, --> JSON.parse('1') = `number` not string, maybe parse, and then stringify after? :dunno: